src/storage/mod.rs

```diff
@@ -339,6 +352,14 @@ pub trait Engine {
     fn create_table(&mut self) -> Result<(), DatabaseError>;
 }
 
+#[repr(u8)]
+#[derive(Clone,Copy,Debug,RustcDecodable, RustcEncodable)]
+enum EngineID {
+    FlatFile = 1,
+    InvertedIndex = 2,
```

Comment:
Enumeration begins with index 1 all other items will be automatically incremented